HDF5 Build Fails to Incorporate Zlib and and SZIP

### Description / Steps to reproduce the issue

Here is a test program `test_filters.c`:
```
#include "hdf5.h"
#include <stdio.h>

int main() {
        H5open();
        printf("H5Z DEFLATE filter: %d\n", H5Zfilter_avail(1));
        printf("H5Z SZIP filter: %d\n", H5Zfilter_avail(4));
        return 0;
}
```

Compiling this with `gcc test_filters.c -lhdf5 -o test_filters` and running `test_filters` results in the following output:
```
$ ./test_filters
H5Z DEFLATE filter: 1
H5Z SZIP filter: 0
```

Running `make-pkg` gives the following output:
```
==> Starting build()...
loading initial cache file ../hdf5-{1.14.2}/config/cmake
CMake Error: Error processing file: D:/repos/MINGW-packages/mingw-w64-hdf5/src/hdf5-{1.14.2}/config/cmake
-- Final: share/doc/hdf5
-- Testing maximum decimal precision for C - 21;33
-- maximum decimal precision for C var - 33
-- ... disable color and URL extended diagnostic messages
-- ....All Warnings are enabled
-- Could NOT find ZLIB (missing: ZLIB_DIR)
-- Could NOT find libaec (missing: libaec_DIR)
-- Could NOT find SZIP (missing: SZIP_DIR)
CMake Warning at CMakeFilters.cmake:147 (find_package):
  By not providing "FindSZIP.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "SZIP", but
  CMake did not find one.

  Could not find a package configuration file provided by "SZIP" with any of
  the following names:

    SZIPConfig.cmake
    szip-config.cmake

  Add the installation prefix of "SZIP" to CMAKE_PREFIX_PATH or set
  "SZIP_DIR" to a directory containing one of the above files.  If "SZIP"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:835 (include)


CMake Warning at CMakeFilters.cmake:183 (message):
  SZIP support in HDF5 was enabled but not found
Call Stack (most recent call first):
  CMakeLists.txt:835 (include)
```

### Expected behavior

```
$ ./test_filters
H5Z DEFLATE filter: 1
H5Z SZIP filter: 1
```


### Actual behavior

```
$ ./test_filters
H5Z DEFLATE filter: 1
H5Z SZIP filter: 0
```


### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19044

### MINGW environments affected

- [X] MINGW64
- [X] MINGW32
- [X] UCRT64
- [X] CLANG64
- [X] CLANG32
- [X] CLANGARM64

### Are you willing to submit a PR?

Yes, but I need help.

--- a/mingw-w64-hdf5/PKGBUILD
+++ b/mingw-w64-hdf5/PKGBUILD
@@ -7,7 +7,7 @@ pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 _ver=1.14.2
 patch=
 pkgver=${_ver}${patch//-/.}
-pkgrel=1
+pkgrel=2
 pkgdesc="General purpose library and file format for storing scientific data (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
@@ -71,7 +71,6 @@ build() {
 
   CFLAGS+=" -Wno-implicit-function-declaration" \
   MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
-  SZIP_INSTALL=${MINGW_PREFIX} \
     ${MINGW_PREFIX}/bin/cmake \
       -Wno-dev \
       -GNinja \
@@ -86,7 +85,9 @@ build() {
       -DHDF5_BUILD_TOOLS=ON \
       -DHDF5_ENABLE_DEPRECATED_SYMBOLS=ON \
       -DHDF5_ENABLE_SZIP_SUPPORT=ON \
+      -Dlibaec_DIR=${MINGW_PREFIX}/lib/cmake \
       -DHDF5_ENABLE_Z_LIB_SUPPORT=ON \
+      -DZLIB_ROOT=${MINGW_PREFIX} \
       -DHDF5_ENABLE_ROS3_VFD=ON \
       -DHDF5_INSTALL_CMAKE_DIR="lib/cmake" \
       -DHDF5_INSTALL_DATA_DIR="share/doc/hdf5" \

